File: Scarab-0.1.00d19/perl/echo-server.pl
use lib 'lib';

use IO::Socket;
use RPC::Scarab::Manager;

my $object = Echo->new;

my $server = RPC::Scarab::Manager->new('root' => $object);

print "connect to me on port " . $server->receive_port->sockport . "\n";

$server->run_loop;


######################################################################
###
### This is the class that the client talks to
###

# Echo has one method, `echo'.  `echo' returns it's first argument
# back to the caller.

package Echo;

sub new {
    my $type = shift;

    bless {}, $type;
}

sub echo {
    my $self = shift;
    my $string = shift;

    return $string;
}

File: supporting_libs/scarab/Scarab-0.1.00d19/python/echo-server.py
# Echo has one method, `echo'.  `echo' returns it's first argument
# back to the caller.

class Echo:

	def echo(self, string):
		return string

# Echo server program
import Scarab

server = Scarab.server_on()
server.globals['root'] = Echo()

print "Server listening on: " + server.url

server.run_loop()
